### DISCUSSION TITLE
Merging bands

### DISCUSSION THREAD

**Discussion by Ixiodor** [corteva/rioxarray#200] [Q&A]
I'm trying to merge N bands from N files.
Reading single files results in N array with shape (1, X, Y) that I'm trying to merge with :
`merged = xarray.merge([b2, b3, b4, b8]).`
That become a Dataset that it's unable to be converted to tiff trought:
`merged.rio.to_raster("output.tif")`

I suspect it's cause single bands merged are 3D arrays(default by `rioxarray.open_rasterio`).
How i can properly merge single bands in one 3D array and save them as a GEOTIFF with multiple bands?

**Comment by snowman2** [comment]
@Ixiodor are you attempting to do a geospatial merge? Also, are you able to provide the error messages you are seeing and the input data?

**Reply by Ixiodor** [reply]
Yes, they are geospatial, georeferenced data.
My error says that  `merged.rio.to_raster("output.tif")` can only be done one 2D or 3D arrays.
Naming each array(band) i get a dataset with "band_N : shape(1, X, Y)".

For now, seems to solve this `merged = xarray.concat([b2,b3,b4,b8], dim="band")`. Uses the z dimentions(band) to merge over.

**Reply by snowman2** [reply]
If you want it merged geospatially, I would recommend: https://corteva.github.io/rioxarray/stable/examples/merge.html

**Reply by Ixiodor** [reply]
`merged = merge_arrays(arrays)`
That fail(error I that said before) when you try to save as geotiff if dataset bands are in shape (1, X, Y).
To be more precise:
`merged = merge_arrays(arrays)` succced but you get no change, your result will be a (1, X, Y) DataArray.

**Reply by snowman2** [reply]
If you use `.squeeze()` or `isel(band=0)` it should work.
